template/clients/rust/scripts/client/format-rust.mjs

```diff
@@ -0,0 +1,28 @@
+#!/usr/bin/env zx
+import 'zx/globals';
+import {
+  cliArguments,
+  getToolchainArgument,
+  popArgument,
+  workingDirectory,
+} from '../utils.mjs';
+
+// Configure additional arguments here, e.g.:
+// ['--arg1', '--arg2', ...cliArguments()]
+const formatArgs = cliArguments();
+
+const fix = popArgument(formatArgs, '--fix');
+const toolchain = getToolchainArgument('format');
+const manifestPath = path.join(
+  workingDirectory,
+  'clients',
+  'rust',
+  'Cargo.toml'
+);
+
+// Format the client.
+if (fix) {
+  await $`cargo ${toolchain} fmt --manifest-path ${manifestPath} -- ${formatArgs}`;
+} else {
+  await $`cargo ${toolchain} fmt --manifest-path ${manifestPath} -- --check ${formatArgs}`;
+}
```

Comment:
Here's one limitation to be aware of with these scripts and Cargo. For commands like `cargo fmt`, and other commands that cargo provides a wrapper for, there's a way to provide arguments to `cargo` and/or arguments to the subcommand, say `rustfmt`.

The delimiter is `--`. For example, consider the command.
```
cargo fmt --manifest-path <a> -- --check <b>
```

The`<a>` is an option for `cargo`, since it comes before `--`, while `<b>` is for `rustfmt`.

We currently are limiting these `cliArguments()` to be only for `rustfmt` here. Personally I think it's fine for now, but just wanted to call it out.

Reply:
Good call!

I think it may be confusing to leave it as-is since people might add arguments without understanding they have already left the realm of `cargo` options.

If we went with position `<a>` in your example for `cliArguments()`, and someone provided the following options `--foo -- --bar`, we would end up with:

```sh
cargo fmt --manifest-path --foo -- --bar -- --check
```

Would this be an issue because of the double `--`, or is the last one simply ignored?

Reply:
> Would this be an issue because of the double --, or is the last one simply ignored?

Yeah unfortunately this is an error. I think in rustfmt's case it lets you provide another `--` and then the expected argument is a file, but typically you only want one `--` in there.

You can probably tell our major issue here with cargo fmt is the fact that we need to provide `--check` to rustfmt _after_ the delimiter.

Reply:
We could get real fancy here and just inspect the CLI args for a `--` followed somewhere by a `--check`, and if that's not matched, add the `--check` flag? I don't think it would be too nasty to set up.

Reply:
Sounds good. I think that's the last bit for me, then this can ship!

Reply:
I've made [this change](https://github.com/solana-program/create-solana-program/pull/65/commits/9ea44b4c99b78992e78fde00fcda4d696a069d6f) and tested in on [the demo PR](https://github.com/solana-program/counter-shank/pull/1). Let me know what you think. 🙂

Reply:
Hell yeah I think that's perfect!